airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -238,6 +242,8 @@ private void replicate(final Path jobRoot,
                 replicationRunnableFailureRef.set(FailureHelper.sourceFailure(ex, Long.valueOf(jobId), attempt));
               } else if (ex.getCause() instanceof DestinationException) {
                 replicationRunnableFailureRef.set(FailureHelper.destinationFailure(ex, Long.valueOf(jobId), attempt));
+              } else if (ex.getCause() instanceof HeartbeatTimeoutChaperone.HeartbeatTimeoutException) {
```

Comment:
Nit:  it feels like this if/else block is due to be refactored into a method that returns the FailureHelper.  It would allow us to write a test to verify the behavior by itself and explain what it is doing.

Reply:
Done